### ISSUE TITLE
Just starting out with a small headache

### ISSUE THREAD

**Issue by Ryokoie** [Thutmose/ThutPerms#18]
Hi there! So I am having the biggest headache in a half. I have tried Bukkit, Spigot, Sponge. I can't get them to work with me. I seen this as a god send. However, while assigning permissions to a group. That group can't use said permissions without me enabling them to be able to use Creative Mode, Op, etc. 

I am currently running forge 1.12.2 - 2655? I believe. Along with the most recent version of this mod. I have java all the way up to date.  I am using the mod for a Pixelmon server, and when a player types in a command. It doesn't do anything and I get the following messages in the log:

"[13:10:33] [Server thread/ERROR] [FML]: Exception caught during firing event net.minecraftforge.event.CommandEvent@71ee8fed:
java.lang.NullPointerException: null
	at thut.permissions.PermissionsHolder.hasPermission(PermissionsHolder.java:39) ~[PermissionsHolder.class:?]
	at thut.permissions.GroupManager.hasPermission(GroupManager.java:85) ~[GroupManager.class:?]
	at thut.permissions.ThutPerms.canUse(ThutPerms.java:257) ~[ThutPerms.class:?]
	at thut.permissions.ThutPerms.commandUseEvent(ThutPerms.java:136) ~[ThutPerms.class:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_144_ThutPerms_commandUseEvent_CommandEvent.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) ~[ASMEventHandler.class:?]
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182) [EventBus.class:?]
	at net.minecraft.command.CommandHandler.func_71556_a(CommandHandler.java:54) [bj.class:?]
	at net.minecraft.network.NetHandlerPlayServer.func_147361_d(NetHandlerPlayServer.java:958) [pa.class:?]
	at net.minecraft.network.NetHandlerPlayServer.func_147354_a(NetHandlerPlayServer.java:937) [pa.class:?]
	at net.minecraft.network.play.client.CPacketChatMessage.func_148833_a(SourceFile:37) [la.class:?]
	at net.minecraft.network.play.client.CPacketChatMessage.func_148833_a(SourceFile:9) [la.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) [hv$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_151]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_151]
	at net.minecraft.util.Util.func_181617_a(SourceFile:46) [h.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:721) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:396) [nz.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:666) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:524) [MinecraftServer.class:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_151]
[13:10:33] [Server thread/ERROR] [FML]: Index: 2 Listeners:
[13:10:33] [Server thread/ERROR] [FML]: 0: NORMAL
[13:10:33] [Server thread/ERROR] [FML]: 1: ASM: com.sk89q.worldedit.forge.ForgeWorldEdit@6cf32e65 onCommandEvent(Lnet/minecraftforge/event/CommandEvent;)V
[13:10:33] [Server thread/ERROR] [FML]: 2: ASM: thut.permissions.ThutPerms@72ce263c commandUseEvent(Lnet/minecraftforge/event/CommandEvent;)V
[13:10:33] [Server thread/ERROR] [FML]: 3: LOWEST
[13:10:33] [Server thread/ERROR] [FML]: 4: ASM: thut.essentials.commands.chatcontrol.Spy@79929120 chat(Lnet/minecraftforge/event/CommandEvent;)V
[13:10:33] [Server thread/FATAL] [net.minecraft.server.MinecraftServer]: Error executing task
java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(Unknown Source) ~[?:1.8.0_151]
	at java.util.concurrent.FutureTask.get(Unknown Source) ~[?:1.8.0_151]
	at net.minecraft.util.Util.func_181617_a(SourceFile:47) [h.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:721) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:396) [nz.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:666) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:524) [MinecraftServer.class:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_151]
Caused by: java.lang.NullPointerException
	at thut.permissions.PermissionsHolder.hasPermission(PermissionsHolder.java:39) ~[PermissionsHolder.class:?]
	at thut.permissions.GroupManager.hasPermission(GroupManager.java:85) ~[GroupManager.class:?]
	at thut.permissions.ThutPerms.canUse(ThutPerms.java:257) ~[ThutPerms.class:?]
	at thut.permissions.ThutPerms.commandUseEvent(ThutPerms.java:136) ~[ThutPerms.class:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_144_ThutPerms_commandUseEvent_CommandEvent.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) ~[ASMEventHandler.class:?]
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182) ~[EventBus.class:?]
	at net.minecraft.command.CommandHandler.func_71556_a(CommandHandler.java:54) ~[bj.class:?]
	at net.minecraft.network.NetHandlerPlayServer.func_147361_d(NetHandlerPlayServer.java:958) ~[pa.class:?]
	at net.minecraft.network.NetHandlerPlayServer.func_147354_a(NetHandlerPlayServer.java:937) ~[pa.class:?]
	at net.minecraft.network.play.client.CPacketChatMessage.func_148833_a(SourceFile:37) ~[la.class:?]
	at net.minecraft.network.play.client.CPacketChatMessage.func_148833_a(SourceFile:9) ~[la.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) ~[hv$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:1.8.0_151]
	at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:1.8.0_151]
	at net.minecraft.util.Util.func_181617_a(SourceFile:46) ~[h.class:?]
	... 5 more"

All I work like to do is have the group be able to use Ivs, Evs, Wondertrade, Spawn, TPA, the basics. Would anyone be able to help me out? I would greatly appreciate it.

**Comment by Thutmose** [comment]
this is a duplicate of #15, I will get it fixed whenever i have time, which, hopefully will be sometime after thursday, until then, I am very busy with some other things, so don't have time to look into minecraft stuff.

**Comment by Ryokoie** [comment]
Ah gotcha, alright then. Thank you very much.

**Comment by Thutmose** [comment]
the past few months I have been very busy, and not had any time to get much done, after thursday, I will go through my backlog of bugs in the following order:

Pokecube
Elevators
ThutPerms/ThutEssentials
Everything else

I will go through the issues on all of them, and get them cleared out (bug report ones, not suggestion ones), then once all the bug reports on one is done, will move to next on the list, and then might go look at suggestion issues.

**Comment by Thutmose** [comment]
Can you test this version here? 
[thutpermissions-1.12.2-2.0.4.jar.zip](https://github.com/Thutmose/ThutPerms/files/1977192/thutpermissions-1.12.2-2.0.4.jar.zip)

the jar is in a zip so github allows uploading it to a comment

**Comment by CapturedCherry-zz** [comment]
I still have the same issues using thutpermissions-1.12.2-2.0.4.jar.

[22:47:18] [Server thread/ERROR] [FML]: Exception caught during firing event net.minecraftforge.event.CommandEvent@37a3f32b:
java.lang.NullPointerException: null
	at thut.permissions.PermissionsHolder.hasPermission(PermissionsHolder.java:38) ~[PermissionsHolder.class:?]
	at thut.permissions.GroupManager.hasPermission(GroupManager.java:85) ~[GroupManager.class:?]
	at thut.permissions.ThutPerms.canUse(ThutPerms.java:257) ~[ThutPerms.class:?]
	at thut.permissions.ThutPerms.commandUseEvent(ThutPerms.java:136) ~[ThutPerms.class:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_37_ThutPerms_commandUseEvent_CommandEvent.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) ~[ASMEventHandler.class:?]
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182) [EventBus.class:?]
	at net.minecraft.command.CommandHandler.func_71556_a(CommandHandler.java:54) [bj.class:?]
	at net.minecraft.network.NetHandlerPlayServer.func_147361_d(NetHandlerPlayServer.java:958) [pa.class:?]
	at net.minecraft.network.NetHandlerPlayServer.func_147354_a(NetHandlerPlayServer.java:937) [pa.class:?]
	at net.minecraft.network.play.client.CPacketChatMessage.func_148833_a(SourceFile:37) [la.class:?]
	at net.minecraft.network.play.client.CPacketChatMessage.func_148833_a(SourceFile:9) [la.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) [hv$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_162]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_162]
	at net.minecraft.util.Util.func_181617_a(SourceFile:46) [h.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:721) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:396) [nz.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:666) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:524) [MinecraftServer.class:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_162]
[22:47:18] [Server thread/ERROR] [FML]: Index: 1 Listeners:
[22:47:18] [Server thread/ERROR] [FML]: 0: NORMAL
[22:47:18] [Server thread/ERROR] [FML]: 1: ASM: thut.permissions.ThutPerms@359a53f5 commandUseEvent(Lnet/minecraftforge/event/CommandEvent;)V
[22:47:18] [Server thread/ERROR] [FML]: 2: LOWEST
[22:47:18] [Server thread/ERROR] [FML]: 3: ASM: thut.essentials.commands.chatcontrol.Spy@49ca2fde chat(Lnet/minecraftforge/event/CommandEvent;)V
[22:47:18] [Server thread/FATAL] [net.minecraft.server.MinecraftServer]: Error executing task
java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(Unknown Source) ~[?:1.8.0_162]
	at java.util.concurrent.FutureTask.get(Unknown Source) ~[?:1.8.0_162]
	at net.minecraft.util.Util.func_181617_a(SourceFile:47) [h.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:721) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:396) [nz.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:666) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:524) [MinecraftServer.class:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_162]
Caused by: java.lang.NullPointerException
	at thut.permissions.PermissionsHolder.hasPermission(PermissionsHolder.java:38) ~[PermissionsHolder.class:?]
	at thut.permissions.GroupManager.hasPermission(GroupManager.java:85) ~[GroupManager.class:?]
	at thut.permissions.ThutPerms.canUse(ThutPerms.java:257) ~[ThutPerms.class:?]
	at thut.permissions.ThutPerms.commandUseEvent(ThutPerms.java:136) ~[ThutPerms.class:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_37_ThutPerms_commandUseEvent_CommandEvent.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) ~[ASMEventHandler.class:?]
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182) ~[EventBus.class:?]
	at net.minecraft.command.CommandHandler.func_71556_a(CommandHandler.java:54) ~[bj.class:?]
	at net.minecraft.network.NetHandlerPlayServer.func_147361_d(NetHandlerPlayServer.java:958) ~[pa.class:?]
	at net.minecraft.network.NetHandlerPlayServer.func_147354_a(NetHandlerPlayServer.java:937) ~[pa.class:?]
	at net.minecraft.network.play.client.CPacketChatMessage.func_148833_a(SourceFile:37) ~[la.class:?]
	at net.minecraft.network.play.client.CPacketChatMessage.func_148833_a(SourceFile:9) ~[la.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) ~[hv$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:1.8.0_162]
	at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:1.8.0_162]
	at net.minecraft.util.Util.func_181617_a(SourceFile:46) ~[h.class:?]
	... 5 more

**Comment by Thutmose** [comment]
can you send a full list of every mod you have installed? if it actually crashed, sending the crash report that it generates would work as well, as that includes a modlist

**Comment by CapturedCherry-zz** [comment]
thutessentials-1.12.2-2.1.6 and thutpermissions-1.12.2-2.0.4

**Comment by Thutmose** [comment]
and those are the only things inside the mods folder? what version of minecraft forge?

**Comment by CapturedCherry-zz** [comment]
Forge version 14.23.3.2676
Minecraft Version 1.12.2

**Comment by DarkmoreSOS** [comment]
![image](https://user-images.githubusercontent.com/28847909/39678277-37eb7f14-5181-11e8-8917-ee96ec1baee7.png)

**Comment by Thutmose** [comment]
I will make a new server on my end for testing then, with:
Minecraft Forge 2676,
thutessentials-1.12.2-2.1.6
thutpermissions-1.12.2-2.0.4

What command had you run to cause the crash? and from where?

**Comment by CapturedCherry-zz** [comment]
I'm so sorry I should had gave you more info my bad. 
Version thutpermissions-1.12.2-2.0.4 fix the crash but the now when you type any command like /help in-game that's the error I get in console. But when I set "all" to true for the default group the commands work but players have accuse to all commands. 
I will set the other groups.

**Comment by Thutmose** [comment]
Can you join this discord? https://discord.gg/medHvkA it will be easier for discussing some things there, I am not able to get anything to go wrong with the permissions on my end, it all seems to be working fine, so there is obviously something different that we are doing, and I am not sure what it is

**Comment by CapturedCherry-zz** [comment]
I have joined. Were do I put my issues?

**Comment by Thutmose** [comment]
this should be fixed as of the 2.0.4 which has been uploaded to curseforge (not the one linked further up this thread)
